packages/aws-cdk-lib/aws-iam/lib/private/imported-role.ts

```diff
@@ -74,8 +74,15 @@ export class ImportedRole extends Resource implements IRole, IComparablePrincipa
     }
   }
 
-  public addManagedPolicy(policy: IManagedPolicy): void {
-    Annotations.of(this).addWarningV2('@aws-cdk/aws-iam:importedRoleManagedPolicyNotAdded', `Not adding managed policy: ${policy.managedPolicyArn} to imported role: ${this.roleName}`);
+  public addManagedPolicy(policy: ManagedPolicy): void {
+    try {
+      policy.attachToRole(this);
```

Comment:
We should not change the signature (IManagedPolicy -> ManagedPolicy) because technically it is a breaking change.

Instead I'd write something like `(policy as ManagedPolicy).attachToRole`.

Or use type predicate:

```ts
    const isManagedPolicy = (x: IManagedPolicy): x is ManagedPolicy => {
      return (x as any).attachToRole !== undefined;
    };
    if (isManagedPolicy(policy)) {
      policy.attachToRole(this);
    }
```

This way we don't need catch the error for type check.

Reply:
Resolved with **type predicate** and changed as **Warning** instead of **Error**.
Removed ```try ... catch```